### ISSUE TITLE
Staging Review Healthcare Experience - Questionnaires (AKA: Clipboard) 

### ISSUE THREAD

**Issue by leahkeeler** [department-of-veterans-affairs/va.gov-team#24145]
## Steps to complete Staging Review: 

- [ ] VFS Product Manager: Create this issue and fill in your team and feature name in the title.
- [ ] VFS: Link to this issue once created in [#vfs-platform-support](https://dsva.slack.com/channels/vfs-platform-support) in Slack; tag @vsp-product-support-members. 
- [x] VSP: Schedule meeting with VSP reviewers and requesting team attendees (as listed below)
- [ ] VFS: At least 4 days before scheduled meeting, you must complete list of artifacts below
- [ ] VFS: Verify that access information has been provided for the tool or feature in staging. Do not put staging credentials in your va.gov-team ticket; store or reference them in a .md file in the va.gov-team-sensitive repository.
- [x] VSP: Create feedback tickets and link to the Staging Review ticket (if no feedback, please comment stating that). 
- [ ] VFS: Respond to VSP feedback tickets.
- [ ] VFS: Update this ticket with the product’s launch date after it has gone live.
- [x] VSP: Update [Platform Collaboration Point Tracker](https://docs.google.com/spreadsheets/d/1OgPyEvUlNF6EnaYMFAXJkV6FKOvZnlPnbOQ2fAJ7W7A/edit#gid=266151061).
- [ ] VFS: Close this ticket when the review is complete.

## Artifacts
- Link to [[product outline]](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/questionnaire/product/product-outline.md)
- Link to the[ tool ](https://staging.va.gov/?next=%2Fhealth-care%2Fhealth-questionnaires%2Fquestionnaires)
-  Staging test-user information (if applicable) va.api.user+idme.101@gmail.com 
-  Link to content brief (if applicable)
-  Link to or specify the [artifact](https://scene.zeplin.io/project/5f3d86a458ea2231e60a4780) that represents this initiative's content source of truth 
- Link to [release plan](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/product-management/release-plan-template.md) with the "Planning" sections completed (in each section: Phase I, Phase II, Go Live)
- Link to your product test [cases](https://dsvavsp.testrail.io/index.php?/suites/view/35&group_by=cases:section_id&group_order=asc&display_deleted_cases=0&group_id=896)/test [plan](https://dsvavsp.testrail.io/index.php?/plans/view/1215) in TestRail with test execution logs ([learn more](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/quality-assurance/qa-artifacts.md#test-plan)) _for QA practice area review_
- Link to your [Coverage for References](https://dsvavsp.testrail.io/index.php?/reports/view/223) and [Summary(Defects)](https://dsvavsp.testrail.io/index.php?/reports/view/222) reports in TestRail _for QA practice area review_
- Link to verification that accessibility checks are complete:
  - (preferred) Link to your [TestRail VSP accessibility test plan template](https://dsvavsp.testrail.io/index.php?/suites/view/14&group_by=cases:section_id&group_order=asc).
OR
  - Link to a doc in your product folder that indicates all required checks from the accessibility [staging review prep](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/staging-review-processes.md) doc are complete.
- List of known issues (with links to tickets when applicable)

## Meeting attendees from requesting team
- Product Manager (required): Leah Keeler (Interim) 
- DEPO Product lead (recommended): Ming Ligh
- DEPO Design lead (required): Ryan Thurlwell
- Anyone else from your team who significantly contributed to the artifacts provided (required): Kristen McConnel, Mark Dewey
- Content, IA, accessibility, and QA specialists (if applicable): name(s) Tze Lei

## Scheduling
What is your preferred time slot per the [VSP Collaboration Meetings Calendar] **5/13/2021 1:30 PM** (https://calendar.google.com/calendar/u/0/embed?src=adhocteam.us_4dn3o77gcm5e3vbiedlha96tc0@group.calendar.google.com&ctz=America/New_York)? **date/time**
 
Please note:
- All times are shown in ET. 
- Available time slots are indicated with [CALENDAR BLOCK]. 
- When you reserve a time slot, the reserved meeting will show up alongside the [CALENDAR BLOCK]. 
- Please select a time slot at least 4 business days from request date.

## Additional information
Please refer to the [VSP Collaboration Cycle README](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/working-with-vsp/vsp-collaboration-cycle/README.md) on Github for more information about the Collaboration Cycle.

**Comment by shiragoodman** [comment]
@leahkeeler scheduled for 5/13 at 1:30 pm ET. Also, I changed the name since my team has known this product as "Questionnaires"  and I don't want to cause confusion.

**Comment by tlei123** [comment]
@pjhill @ddzz , all QA test-runs now completed, and QA-artifact links updated above in main description.

cc @leahkeeler @shiragoodman

**Comment by leahkeeler** [comment]
@allison0034 Unfortunately we cannot get more - however the test users can be reset, if needed @mdewey can reset, but it needs to be coordinated.

**Comment by mnorthuis** [comment]
@leahkeeler Is there a ticket for the static landing page and entry point work through Sitewide content?  Also, is there copy established for appointment texts and emails that will include links to this tool?

**Comment by AngelaFowler82** [comment]
Can we reset user 101? Also, would someone kindly send me an invite to the staging review? Huge thanks.

**Comment by kristenmcconnell** [comment]
Here is what the text will look like. The VeText team has been working on the configurations. 
<img width="321" alt="Screen Shot 2021-05-11 at 4 05 42 PM" src="https://user-images.githubusercontent.com/66287082/117877715-e3a21c00-b272-11eb-9e01-821ffd0601fd.png">

**Comment by AngelaFowler82** [comment]
Sadly, it is still saying it has expired. I tried in two different browsers in case it was a cash issue with Chrome.

**Comment by shiragoodman** [comment]
@AngelaFowler82 please reach out to Mark Dewey @mdewey on Slack and he can help you out.

**Comment by mnorthuis** [comment]
@kristenmcconnell One of the flows we had originally discussed, was for users who bookmark or deep link into a specific questionnaire ([referenced in flows here as #2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/information-architecture/ia-reviews/health-questionaires.md)), to be taken to an unauthenticated view of that questionnaire, they would then be prompted to sign in, and then taken either to the list page or the specific questionnaire depending on its availability.  My understanding was that this was in prep for the texts/emails to eventually also deep link users directly to specific questionnaires without being authenticated.   

For the expired survey that is out there, the deep link to it is: https://staging.va.gov/health-care/health-questionnaires/questionnaires/answer-questions/introduction?id=I2-3PYJBEU2DIBW5RZT2XI3PASYGN6Q3BYXUYBP5LKE7PZGP6ZFS7RA0000
Here's what happens:   It sends you to the login process, then loads/flashes the home page, then loads the questionnaire list page.   There is quite a bit of screen reloading in this flow that accessibility should look at, but there is no unauthenticated version of the questionnaire.   

Can you confirm that the approach for this flow was changed intentionally or if this is an error?

For a survey that is available to complete: https://staging.va.gov/health-care/health-questionnaires/questionnaires/answer-questions/introduction?id=I2-3PYJBEU2DIBW5RZT2XI3PASYGMQFEQEGUDEHYKKFBLHSVPEO6XJA0000
Deep linking as an unauthenticated user to this survey goes through quite a bit of screen loading after sign in as well, but does allow the user to land on the specific questionnaire. 

@AngelaFowler82 if you have time, could you review one of these flows to ensure the amount of screen loading and reloading is not an issue?

**Comment by mdewey** [comment]
I can answer the unauthenticated flow questions.

This is for the unauthenticated flow, where if a user would leave a questionnaire that they bookmarked, left the page, and came back.

TL;DR; It ended not being needed/possible, so we punted on that for future iterations. 

Long story: 

After a user leaves the site, the data for the appointment and questionnaire is no longer client side. Even for an un-authenticated form, we would have to load data to display the appointment details (what type of questionnaire, the facility, the appointment details, etc.).

To load the data to display the form in the correct state (new vs. returning) and the correct appointment details, we need to know which patient is visiting the questionnaire. We do not know the patient that is visiting if the user is not authenticated. We decided that if the user is not logged in to redirect to the login process and then to the questionnaire page. For the pilot, we assume a small number of questionnaires for a user, so the overhead of finding their questionnaire was low.

Until the (new) check-in project came up, we would improve that flow and provide better deep linking, but I am not sure on the priority from the team now that resources are being pivoted away. We do have thoughts and ideas to improve the flow, but they are outside of the scope of the MVP. 

Let me know if you have any questions :-)

**Comment by mnorthuis** [comment]
Thanks @mdewey!  Just want to update my notes in regards to that flow change.

**Comment by ddzz** [comment]
@leahkeeler @tlei123 @shiragoodman @pjhill  The ID.me account included in the TestRail test plan only has an expired questionnaire. I tried using a few other testing accounts, but they do not have any questionnaires. Are there any accounts that have a form that can be completed? We can't do a full QA review without a working questionnaire.

**Comment by mdewey** [comment]
@ddzz  I have to go in and reset the data for you. I will go in a reset it now.

**Comment by leahkeeler** [comment]
Hi @ddzz Pink @mdewey in the Clipboard slack and he can reset them for you - sorry, it's kind of a manual process with how the test users are set up

**Comment by mdewey** [comment]
@ddzz I have reset the data for the test user. You should see a nonexpired questionnaire. Let me know if you need anything else.

**Comment by jparker0425** [comment]
Item remaining for PDF has been logged in another item. 

Launch date of 6/4.
